src/bionics.cpp

```diff
@@ -639,6 +780,275 @@ void player::power_bionics()
+            ( tab_index > 0 ) ? tab_index-- : tab_index = tab_count - 1;
+            cursor = 1;
+            scroll_position = 0;
+            recalc = true;
+
+        } else if( action == "RIGHT" ) {
+            tab_index = ( tab_index + 1 ) % tab_count;
+            cursor = 1;
+            scroll_position = 0;
+            recalc = true;
+
+        } else if( action == "REASSIGN" ) {
+            const long ch = my_bionics[content[cursor].second].invlet;
+            bionic *tmp = bionic_by_invlet( ch );
+            if( tmp == nullptr ) {
+                // Selected an non-existing bionic (or escape, or ...)
+                continue;
+            }
+
+            const long newch = popup_getkey( _( "%s; enter new letter." ),
+                                             bionics[tmp->id].name.c_str() );
+            wrefresh( w_bionics );
+            if( newch == ch || newch == ' ' || newch == KEY_ESCAPE ) {
+                continue;
+            }
+            if( !bionic_chars.valid( newch ) ) {
+                popup( _( "Invalid bionic letter. Only those characters are valid:\n\n%s" ),
+                       bionic_chars.get_allowed_chars().c_str() );
+                continue;
+            }
+            bionic *otmp = bionic_by_invlet( newch );
+            if( otmp != nullptr ) {
+                std::swap( tmp->invlet, otmp->invlet );
+            } else {
+                tmp->invlet = newch;
+            }
+
+        } else if( action == "REMOVE" ) {
+            if( uninstall_bionic( my_bionics[content[cursor].second].id ) ) {
+                recalc = true;
+                continue;
+            }
+
+        } else if( action == "CONFIRM" ) {
+            bionic *tmp = &my_bionics[content[cursor].second];
+            if( tmp == nullptr ) {
+                // Selected an non-existing bionic
+                // this should not happen because we're choosing bionic via cursor
+                continue;
+            } else {
+                if( bionics[tmp->id].activated ) {
+                    int b = tmp - &my_bionics[0];
+                    if( tmp->powered ) {
+                        deactivate_bionic( b );
+                    } else {
+                        activate_bionic( b );
+                    }
+                    // update message log and the menu
+                    g->refresh_all();
+                    continue;
+                } else {
+                    popup( _("You can not activate %s!" ), bionic_info( tmp->id ).name.c_str() );
+                }
+            }
+
+        } else if( action == "ANY_INPUT" ) {
+            long ch = ctxt.get_raw_input().get_first_input();
+            if( ch == KEY_ESCAPE ) {
+                return;
+            }
+
+            // todo: remove code duplication
+            bionic *tmp = bionic_by_invlet( ch );
+            if( tmp == nullptr ) {
+                // Selected an non-existing bionic
+                continue;
+            } else {
```

Comment:
Btw. you can get rid of some of the indentation here. An `else` after `continue` or `return` is not needed. This works the same:
```C++
if( ... ) {
  continue;
}
// remaining code, formerly in the else branch.
```

You can also use that approach above with the check for an active bionic and invert the condition:
```C++
if( !bionic.activated ) {
  popup(...)
  continue;
}
// actual code for toggling the bionic.
```

----

Anyway, those are only style considerations, not even issues.